artemis-server/src/main/java/org/apache/activemq/artemis/core/persistence/impl/journal/LargeServerMessageImpl.java

```diff
@@ -376,6 +387,11 @@ public synchronized void validateFile() throws ActiveMQException {
             openFile();
 
             bodySize = file.size();
+
+            if (keepFileState) {
```

Comment:
You should have done the opposite...


if you named this variable keep, you should have done if !keepFileState


and you should have sent false when not needed...


if the problem was just the toStringI(), just remove the filesize from the toString()... no need for anything else.